File: 백준/Bronze/2747. 피보나치 수/피보나치 수.c
#include <stdio.h>
#include <stdlib.h>
int fibo(int *memo, int n) {
    if (n < 2)
        return n;
    if (memo[n])
        return memo[n];
    memo[n] = fibo(memo, n - 1) + fibo(memo, n - 2);
    return memo[n];
}
int main(void) {
    int n;
    scanf("%d", &n);
    int *memo = (int *)malloc(sizeof(int) * (n + 1));
    for (int i = 0; i < n + 1; i++)
        memo[i] = 0;
    printf("%d", fibo(memo, n));
    return 0;
}

File: 백준/Bronze/2747. 피보나치 수/피보나치 수.py
import sys
n=int(sys.stdin.readline().rstrip())
memo=[0]*(n+1)
def fibo(n):
    if n<2:
        return n
    if memo[n]:
        return memo[n]
    memo[n]=fibo(n-1)+fibo(n-2)
    return memo[n]
print(fibo(n))